src/components/datepicker/datePicker.scss

```diff
@@ -57,9 +85,11 @@ md-datepicker {
 
 // Floating pane that contains the calendar at the bottom of the input.
 .md-datepicker-calendar-pane {
+  // We need to pull the pane backwards, otherwise it
+  // causes an overflow in IE and Edge.
+  left: -100%;
```

Comment:
[A little bit down](https://github.com/angular/material/pull/8083/files#diff-1be65f67ef78c5d0c4961b3ff6776f45R99) the element's scale is set to 0, which prevents it from overflowing in all browsers, except IE and Edge. The `left: -100%` prevents that overflow. I'll work on the explanation.